packages/SwingSet/tools/bundleTool.js

```diff
@@ -29,18 +30,23 @@ const providedCaches = new Map();
  * @param {{ format?: string, dev?: boolean }} options
  * @param {(id: string) => Promise<any>} loadModule
  * @param {number} [pid]
- * @returns {BundleCache}
+ * @returns {Promise<BundleCache>}
  */
-export const provideBundleCache = (dest, options, loadModule, pid) => {
+export const provideBundleCache = async (dest, options, loadModule, pid) => {
+  await null;
   const uniqueDest = [dest, options.format, options.dev].join('-');
   let bundleCache = providedCaches.get(uniqueDest);
   if (!bundleCache) {
-    bundleCache = makeNodeBundleCache(dest, options, loadModule, pid);
+    bundleCache = await makeNodeBundleCache(dest, options, loadModule, pid);
     providedCaches.set(uniqueDest, bundleCache);
```

Comment:
This looks like more than a cleanup, but rather, a genuine bug fix with potentially different (now correct) behavior. Looking at `makeNodeBundleCache`, I see it is declared `async` and so certainly returns a promise. The old code stored this promise into the `providedCaches` collection, which is almost certainly wrong. As the price of correcting this, the new code also waits a turn before storing it into the cache. Is this a possible race condition?

Afterwards, both old and new code returns a promise for the bundleCache, though the old code mistypes it. As long as the types pass, I'm not concerned about the bug fix to the corrected type. But what about the bug fix to what is stored in the collection?

I am also more than happy to see this bug corrected. I only raise this because the PR description sounds like this PR has no such significant behavioral change.

So, no change suggested to the code. But perhaps the PR description should discuss this possible behavioral change?

Reply:
Thanks for the careful review. I'll note this in the PR description